app/server.php

```diff
@@ -83,6 +84,80 @@ function($request, $response) {
         }
     );
 
+App::get('/todos')
+    ->inject('request')
+    ->inject('response')
+    ->action(
+        function($request, $response) {
+            $path = \realpath('/app/app/todos.json');
+            $data = json_decode(file_get_contents($path), true);
+            $response->json($data);
+        }
+    );
+
+App::post('/todos')
+    ->inject('response')
+    ->param('task', "", new Wildcard(), 'Prefs key-value JSON object.')
+    ->param('is_complete', true, new Wildcard(), 'Tells whether task is complete or not')
+    ->action(
+        function($task, $is_complete, $response) {
+            $id = uniqid();
+            var_dump($id);
+            $path = \realpath('/app/app/todos.json');
+            $data = json_decode(file_get_contents($path), true);
+            $task_entry = ['id' => $id, 'task' => $task, 'is_complete' => $is_complete];
+            array_push($data, $task_entry);
+            $jsonData = json_encode($data, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+            file_put_contents($path, $jsonData);
+            $response->json($data);
+        }
+    );
+
+App::put('/todos/:id')
+    ->inject('response')
+    ->param('id', "", new Wildcard(), 'name of the todo')
+    ->param('task', "", new Wildcard(), 'Prefs key-value JSON object.')
+    ->param('is_complete', true, new Wildcard(), 'Tells whether task is complete or not')
+    ->action(
+        function($id, $task, $is_complete, $response) {
+            $path = \realpath('/app/app/todos.json');
+            $data = json_decode(file_get_contents($path));
+            // var_dump($data);
+            foreach($data as $object){
+                if($object->id == $id){
+                    $object->task = $task;
+                    $object->is_complete = $is_complete;
+                    break;
+                }
+            }
+            // var_dump($data);
+            $jsonData = json_encode($data, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+            file_put_contents($path, $jsonData);
+            $response->json($data);
+        }
+    );
+
+    App::delete('/todos/:id')
+    ->inject('response')
+    ->param('id', "", new Wildcard(), 'id of the todo')
+    ->action(
+        function($id, $response) {
+            $path = \realpath('/app/app/todos.json');
+            $data = json_decode(file_get_contents($path));
```

Comment:
```suggestion
            $data = json_decode(file_get_contents($path), true);
```

Reply:
We need to read the file contents as an associative array by passing the `true` flag to `json_decode`